ci/check-protobuf-stability.sh

```diff
@@ -35,42 +45,52 @@ function check_non_lf_protos() {
   readonly BUF_IMAGE_TMPDIR="$(mktemp -d)"
   trap 'rm -rf ${BUF_IMAGE_TMPDIR}' EXIT
 
-  # The `SYSTEM_PULLREQUEST_TARGETBRANCH` environment variable is defined by
-  # Azure Pipelines; in order to run this script locally, define it beforehand
-  # as the branch being targeted. For example:
-  #
-  #   SYSTEM_PULLREQUEST_TARGETBRANCH=main bash -x ci/check-protobuf-stability.sh
-  #
-  TARGET="${SYSTEM_PULLREQUEST_TARGETBRANCH:-main}"
-  echo "The target branch is '${TARGET}'."
+  echo "Checking protobufs against tag '${BUF_TAG_TO_CHECK}'"
+  for buf_module in "${BUF_MODULES_AGAINST_STABLE[@]}"; do
+    (eval "$(dev-env/bin/dade assist)" ; buf breaking --config "${buf_module}" --against "$BUF_TAG_TO_CHECK")
+  done
 
-  # For `main` and PRs targeting `main`, we simply check against the most recent
-  # stable tag.
-  #
-  # For PRs targeting release branches, we should really check against
-  # all the most recent stable tags reachable from either the current branch or
-  # from previous release branches (say, against both `1.17.1` and `1.16.2`
-  # created after the `release/1.17.x` branch).
-  # Instead, we approximate by checking only against the most recent stable tag
-  # reachable from the current branch, under the assumption that if a lesser
-  # release branch contains a protobuf change, then it will also be present in
-  # higher ones either through a shared commit or a back-port from `main`.
+}
+
+case "${1:---stable}" in
+-h | --help)
+  cat <<USAGE
+Usage: ./check-protobuf-stability.sh [options]
+
+Options:
+  -h, --help: shows this help
+  --stable:     check against the latest stable tag (default)
+  --head:       check against the head commit
+USAGE
+  exit
+  ;;
+--stable)
+  echo "Using stable target"
+  # Check against the most recent stable tag.
   #
-  # Finally, this check does not need to run on release branch commits because
+  # This check does not need to run on release branch commits because
   # they are built sequentially, so no conflicts are possible and the per-PR
   # check is enough.
+  readonly RELEASE_BRANCH_REGEX="^release/.*"
   GIT_TAG_SCOPE=""
   if [[ "${TARGET}" =~ ${RELEASE_BRANCH_REGEX} ]]; then
     GIT_TAG_SCOPE="--merged"
   fi
-
   readonly LATEST_STABLE_TAG="$(git tag ${GIT_TAG_SCOPE} | grep -v "snapshot" | sort -V | tail -1)"
-  echo "Checking protobufs against tag '${LATEST_STABLE_TAG}'"
-  for buf_module in "${BUF_MODULES_AGAINST_STABLE[@]}"; do
-    (eval "$(dev-env/bin/dade assist)" ; buf breaking --config "${buf_module}" --against ".git#tag=${LATEST_STABLE_TAG}")
-  done
-
-}
-
+  BUF_TAG_TO_CHECK=".git#tag=${LATEST_STABLE_TAG}"
+  ;;
+--head)
+  # Check against the head of the target branch.
+  #
+  # This check ensures that backwards compatibility is never broken on the target branch,
+  # which is stricter than guaranteeing compatibility between release tags.
+  echo "Using head target"
+  BUF_TAG_TO_CHECK="https://github.com/digital-asset/daml.git#branch=${TARGET}"
```

Comment:
Not supe important but I'd prefer if the name of the variable (and the message when we print it) did not claim it always contain tags, when in fact it contains buf references to either tags or branches.

Reply:
Good point, renaming this one